cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/services/ProgramLifecycleService.java

```diff
@@ -147,7 +149,8 @@
                           ProvisionerNotifier provisionerNotifier, ProvisioningService provisioningService,
                           ProgramStateWriter programStateWriter, CapabilityReader capabilityReader,
                           ArtifactRepository artifactRepository) {
-    this.maxConcurrentRuns = cConf.getInt(Constants.AppFabric.MAX_CONCURRENT_RUNS);
+    this.maxConcurrentRunning = cConf.getInt(Constants.AppFabric.MAX_CONCURRENT_RUNS, -1);
```

Comment:
Default value should be in cconf, not in code. Just leave out the 2nd parameter.